language-plutus-core/generators/Language/PlutusCore/Generators/Internal/TypeEvalCheck.hs

```diff
@@ -78,6 +82,7 @@ typeEvalCheckBy
     -> TypeEvalCheckM (TermOf TypeEvalCheckResult)
 typeEvalCheckBy eval (TermOf term tbv) = TermOf term <$> do
     let typecheck = annotateTerm >=> typecheckTerm (TypeConfig True mempty defaultTypecheckerGas)
+    _ <- checkValueRestriction term
```

Comment:
You can use `through` from our prelude for composing such functions if you like.

Reply:
I guess I'll keep it the way it's written, but I'm fine either way.